service/history/execution/mutable_state_builder_test.go

```diff
@@ -990,3 +994,274 @@ func (s *mutableStateSuite) buildWorkflowMutableState() *persistence.WorkflowMut
+			currentState: &mutableStateBuilder{
+				executionInfo: &persistence.WorkflowExecutionInfo{
+					WorkflowID:             wfID,
+					DomainID:               domainID,
+					RunID:                  runID,
+					NextEventID:            12,
+					State:                  persistence.WorkflowStateCompleted,
+					CompletionEventBatchID: 10,
+					BranchToken:            []byte("branch-token"),
+				},
+			},
+			historyManagerAffordance: func(historyManager *persistence.MockHistoryManager) {
+				historyManager.EXPECT().ReadHistoryBranch(gomock.Any(), gomock.Any()).Return(nil, errors.New("a transient random error"))
+			},
+
+			expectedResult: nil,
+			expectedErr:    &types.InternalServiceError{Message: "unable to get workflow completion event"},
+		},
+		"A 'transient' internal service error, this should be returned to the caller": {
+			currentState: &mutableStateBuilder{
+				executionInfo: &persistence.WorkflowExecutionInfo{
+					WorkflowID:             wfID,
+					DomainID:               domainID,
+					RunID:                  runID,
+					NextEventID:            12,
+					State:                  persistence.WorkflowStateCompleted,
+					CompletionEventBatchID: 10,
+					BranchToken:            []byte("branch-token"),
+				},
+			},
+			historyManagerAffordance: func(historyManager *persistence.MockHistoryManager) {
+				historyManager.EXPECT().ReadHistoryBranch(gomock.Any(), gomock.Any()).
+					Return(nil, &types.InternalServiceError{Message: "an err"})
+			},
+
+			expectedResult: nil,
+			expectedErr:    &types.InternalServiceError{Message: "an err"},
+		},
+		"initial validation: An invalid starting mutable state should return an error": {
+			currentState: &mutableStateBuilder{
+				executionInfo: &persistence.WorkflowExecutionInfo{},
+			},
+			historyManagerAffordance: func(historyManager *persistence.MockHistoryManager) {},
+			expectedResult:           nil,
+			expectedErr:              &types.InternalServiceError{Message: "unable to get workflow completion event"},
+		},
+	}
+
+	for name, td := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			ctrl := gomock.NewController(t)
+
+			shardContext := shardCtx.NewMockContext(ctrl)
+			shardContext.EXPECT().GetShardID().Return(123).AnyTimes() // this isn't called on a few of the validation failures
+			historyManager := persistence.NewMockHistoryManager(ctrl)
+			td.historyManagerAffordance(historyManager)
+
+			domainCache := cache.NewMockDomainCache(ctrl)
+			domainCache.EXPECT().GetDomainName(gomock.Any()).Return("domain", nil).AnyTimes() // this isn't called on validation
+
+			td.currentState.eventsCache = events.NewCache(shardID, historyManager, config.NewForTest(), log.NewNoop(), metrics.NewNoopMetricsClient(), domainCache)
+			td.currentState.shard = shardContext
+
+			res, err := td.currentState.GetCompletionEvent(context.Background())
+
+			assert.Equal(t, td.expectedResult, res)
+			assert.Equal(t, td.expectedErr, err)
```

Comment:
`assert.EqualError` does better job (I could be wrong. little experience with this assert package)

Reply:
I also admit not being super familiar with that specifically. It looks like it is just doing string comparison though so probably `assert.ErrorAs` and `assert.ErrorIs` I suspect. Let me look